File: src/main/java/com/example/mywindsurfingapp/model/Material.java
package com.example.mywindsurfingapp.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Data;

@Entity
@Data
public class Material {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String subject;
    private String grade;
    private String lessonUnit;
    private String materialType;
    private String language;
    private String userEmail;
    private String generatedContent;
}

File: src/types/material.ts
export interface Material {
  id?: number;
  subject: string;
  grade: string;
  lessonUnit: string;
  materialType: string;
  language: string;
  userEmail: string;
  generatedContent?: string;
  createdAt?: string;
  updatedAt?: string;
  description?: string;
}

// Update MaterialFilters to be more explicit about types
export interface MaterialFilters {
  [key: string]: string | undefined;  // Index signature for any string key
  subject?: string;     // Optional subject filter
  grade?: string;       // Optional grade filter
  lessonUnit?: string;  // Optional lesson unit filter
  materialType?: string;// Optional material type filter
  language?: string;    // Optional language filter
}

export interface PaginatedResponse<T> {
  content: T[];
  totalElements: number;
  totalPages: number;
  size: number;
  number: number;
}

export interface SortConfig {
  field: keyof Material;
  direction: 'ASC' | 'DESC';
}